functional-tests/src/test/groovy/FrontendCore/spec/Permit.groovy

```diff
@@ -11,17 +16,120 @@ import utils.*
 @Title("MDS-MineProfile-PermitTab")
 @Stepwise
 class  PermitSpec extends GebReportingSpec {
+    static PERMIT_NUMBER = "M-"+Const.PERMIT_NUMBER
+
+    @Rule
+    TemporaryFolder dir = new TemporaryFolder()
+
+    @SourceURI
+    URI sourceUri
+    Path scriptLocation = Paths.get(sourceUri)
+
     def setup() {
         to MineProfilePage
+        permitTab.tabSelect.click()
     }
 
-    def "User can view permit"(){
-        when:"I go to the permit tab"
-        permitTab.tabSelect.click()
+    def "User can create a permit"(){
+        when: "I click on the new permit"
+        permitTab.newPermitButton.click()
+        newPermitForm.completePermitForm()
+        then: "A permit with the correct ID is present in the permits tab"
+        permitTab.permitTitle.text() == PERMIT_NUMBER
+    }
+
+    def "User can edit the status of a permit"(){
+        when: "I hover over the the add/edit button."
+        moveToFooterAndHoverOnEdit()
+        permitTab.editPermitStatusButton.click()
+
+        and: "I change the status of the permit to closed."
+        permitTab.editPermitFormStatusDropdown.click()
+        permitTab.closedDropdownOption.click()
+        permitTab.submitEditPermitStatus.click()
+        waitFor() {permitTab.permitRow.children().has(title: "Status").text() != "Open"} 
+    
+        then: "The permit status has changed to closed"
+        permitTab.permitRow.children().has(title: "Status").text() == "Closed"
+    }
+
+    def "User can upload a doc to a permit"(){
+        when: "I open the edit initial permit modal."
+        permitTab.permitRow.children().has(text:"Add/Edit").click()
+        permitTab.openFileModalButton.click()
+        
+        and: "Upload a test file to the permit."
+        def uploadedFile = dir.newFile(Const.TEST_FILE_NAME) << Const.TEST_FILE_CONTENT
+        permitTab.uploadField = uploadedFile.absolutePath
+        permitTab.uploadCompleteMessage
+        permitTab.editPermitFileButton.click()
+        downloadTestFileLink
 
-        then:"I should see a list of permit"
-        assert permitTab.permit_no.text() != ''
+        then: "The file is attatched to the permit."
+        downloadTestFileLink.text() == Const.TEST_FILE_NAME
     }
 
+    def "User can download a doc from a permit"(){
+        when: "The user navigates to the test permit's files"
+        permitTab.permitRow.children().has(text:"Add/Edit").click()
+
+        and: "User downloads the  test file"
+        downloadTestFileLink.click()
+        print(Const.DOWNLOAD_PATH+'/'+Const.TEST_FILE_NAME)
+        def file = new File(Const.DOWNLOAD_PATH+'/'+Const.TEST_FILE_NAME)
+        //allow time for the file to be created in the DOWNLOAD_PATH
+        int counter = 0
+        while (!file.exists()) {
```

Comment:
From what I've seen of `wait`, something like this should work `waitFor { file.exists() }`. Is there a reason for the `while` loop instead?

Reply:
hmm, you may be right, when I wrote this I treated it differently because it was dealing the file system instead of the dom.  Will experiment further.

Reply:
You were right, I've fixed the tenure bit as well to simplify the code.